src/robotFactory.js

```diff
@@ -1,10 +1,96 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  getInfo() {
+    return (
+      `Robot: ${this.name}, `
+      + `Chip version: ${this.chipVersion}, `
+      + `Weight: ${this.weight}`
+    );
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null
+  ) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad) {
+      return 'The drone already has cargo';
+    }
+
+    if (cargo.weight <= this.maxLoadWeight) {
+      this.currentLoad = cargo;
+
+      return 'Cargo is loaded';
+    }
+
+    return `Maximum capacity of this drone is ${this.maxLoadWeight}.
+    The cargo exceeds capacity by ${cargo.weight - this.maxLoadWeight} `;
+  }
+
+  unhookLoad() {
+    if (this.currentLoad === null) {
+      return 'The cargo compartment is already empty';
+    }
```

Comment:
better to check for falseness instead
```suggestion
    if (!this.currentLoad) {
      return 'The cargo compartment is already empty';
    }
```

Reply:
Oh, miss that one